### PR TITLE
VLYBY Bid Adapter: add new bid adapter

### PR DESCRIPTION
**Pull request by thomas-vlyby**
## Type of change
- [ ] Bugfix
- [ ] Feature
- [X]  New bidder adapter  
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

- test parameters for validating bids
```
{
  bidder: 'vlyby',
  params: { 
     publisherId: 'f363eb2b75459b34592cc4',   
       siteId: 'techpreview.vlyby_prebidadapter',
       placement:'default'                      
    }
}
```
- contact email of the adapter’s maintainer
- [x] official adapter submission
- [x] Demo Website with GPT Implementation is [here](https://www.sluga.io/vlyby/prebid/)

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/

### CODE REVIEW

**Review by ChrisHuie — CHANGES_REQUESTED** [review]
@thomas-vlyby can you please add test coverage also for this new bid adapter please

File: modules/vlybyBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { registerBidder } from '../src/adapters/bidderFactory.js'
+
+import { BANNER, VIDEO } from '../src/mediaTypes.js'
+import { config } from '../src/config.js';
+
+const ENDPOINT = '//europe-west3-vlybypoc2019.cloudfunctions.net/prebid';
+const BIDDER_CODE = 'vlyby';
```
Comment (muuki88) [inline modules/vlybyBidAdapter.js:7]: Is there a `GVL_ID` that can be associated with this  bidder? Not sure if this plays well with the `gdprEnforcement` module

File: modules/vlybyBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        },
+        gdprConsent: {
+          consentString: gdprConsent.consentString,
+          gdprApplies: gdprConsent.gdprApplies
+        },
+        bidRequests: validBidRequests.map(({ params, sizes, bidId, adUnitCode }) => ({
+          bidId,
+          adUnitCode,
+          params,
+          sizes
```
Comment (muuki88) [inline modules/vlybyBidAdapter.js:44]: What's the effect of the `sizes` being submitted. Is this only for `BANNER` ads or does the player take them into account too?

File: modules/vlybyBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+import { BANNER, VIDEO } from '../src/mediaTypes.js'
+import { config } from '../src/config.js';
+
+const ENDPOINT = '//europe-west3-vlybypoc2019.cloudfunctions.net/prebid';
+const BIDDER_CODE = 'vlyby';
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [VIDEO, BANNER],
```
Comment (muuki88) [inline modules/vlybyBidAdapter.js:11]: Why banner if this is an outstream implementation?

**Comment by ChrisHuie** [comment]
@thomas-vlyby following up on the requested changes to move forward with this pr

**Comment by muuki88** [comment]
Could be that @thomas-vlyby is not available for a few weeks.

File: modules/vlybyBidAdapter.js (hunk 1 above)
Comment (thomas-vlyby) [reply modules/vlybyBidAdapter.js:7]: @muuki88 There is no Global Vendor List ID. Due to the fact, that we are not processing any user data. For gdprEnforcement. Every Implementation needs a provided user ID and contract between VLYBY and Publisher. So for GPDR we will communicate direct with Publisher. Maybe gdpr.rules[].vendorExceptions should be used in this case.

File: modules/vlybyBidAdapter.js (hunk 2 above)
Comment (thomas-vlyby) [reply modules/vlybyBidAdapter.js:37]: sizes and Banner are special case. For some Publisher we will provide Banner with sizes like 300x250 or 300x600

File: modules/vlybyBidAdapter.js (hunk 3 above)
Comment (thomas-vlyby) [reply modules/vlybyBidAdapter.js:11]: In general this Adapter provides a outstream Mediation for Publishers. They can use this Adapter in AdServer with a price Point from Endpoint. In some cases Publisher can use Banner. This makes this adapter more flexibel, for examaple to add Banner with cheaper eCPM for Fallback etc.

File: modules/vlybyBidAdapter.js (hunk 1 above)
Comment (muuki88) [reply modules/vlybyBidAdapter.js:6]: As long as you don't add any user syncs this is probably fine.

However you are a registered vendor now, so why not use it?

**Comment by thomas-vlyby** [comment]
> @thomas-vlyby can you please add test coverage also for this new bid adapter

@ChrisHuie Test coverage added

**Comment by ChrisHuie** [comment]
@thomas-vlyby we recently pushed Preebid 5.0 and per #6650 we now require bid adapters to at least pass advertiserdomain. Can you please add "adomain" to the response in this pr 🙏

**Comment by thomas-vlyby** [comment]
@ChrisHuie adomain added to response and PR in doc created https://github.com/prebid/prebid.github.io/pull/3064

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**